src/ngx_http_lua_output.c

```diff
@@ -802,4 +802,99 @@ ngx_http_lua_flush_cleanup(void *data)
     ctx->flushing_coros--;
 }
 
+#ifndef NGX_LUA_NO_FFI_API
+//Pre-Condition: offset + len <= size of buf
+int
+ngx_http_lua_ffi_write(ngx_http_request_t *r, const char *buf, size_t offset,
+                       size_t len, char **err)
+{
+    ngx_http_lua_ctx_t          *ctx;
+    ngx_buf_t                   *b;
+    ngx_chain_t                 *cl;
+    ngx_int_t                    rc;
+    const char                  *str_sub;
+
+    if (r == NULL) {
+        *err = "no request object found";
+        return NGX_ERROR;
+    }
+
+    if (buf == NULL) {
+        *err = "no buffer found";
+        return NGX_ERROR;
+    }
+
+    if (len == 0) {
+        *err = "bad len from offset";
+        return NGX_ERROR;
```

Comment:
I think instead of returning an error here, we should just return `NGX_OK` when `len == 0` since it is just a no-op.